src/apify-error.js

```diff
@@ -0,0 +1,16 @@
+export const APIFY_ERROR_NAME = 'ApifyError';
+
+export const INVALID_PARAMETER_ERROR_TYPE = 'INVALID_PARAMETER';
+export const REQUEST_FAILED_ERROR_TYPE = 'REQUEST_FAILED';
+export const REQUEST_FAILED_ERROR_MESSAGE = 'Server request failed.';
+
+export default class ApifyError extends Error {
+    constructor(type, message, details) {
+        super(message);
+        this.name = APIFY_ERROR_NAME;
+        this.type = type;
+        this.details = details;
+
+        Error.captureStackTrace(this, ApifyError);
```

Comment:
1) this way we don't display ApifyError constructor in error stack

2) If we don't set name using `this.name = APIFY_ERROR_NAME` then there will be just `Error` instead of `ApifyError` displayed in error